src/app/core/services/auth.service.ts

```diff
@@ -33,52 +34,36 @@ export class AuthService {
   private usersUrl: string;
   private storageService: StorageService;
   private userSubj: BehaviorSubject<UserAuthModel> = new BehaviorSubject(undefined);
+  private spinner: SpinnerService;
 
   /**
    * Observable<NameModel>
    */
   public user$: Observable<NameModel>;
 
-  constructor(storageService: StorageService, http: HttpClient, @Inject(UsersAPI) usersUrl: string) {
+  constructor(
+    storageService: StorageService,
+    http: HttpClient,
+    @Inject(UsersAPI) usersUrl: string,
+    spinner: SpinnerService,
+  ) {
     this.http = http;
     this.usersUrl = usersUrl;
     this.storageService = storageService;
     this.user$ = this.userSubj.asObservable().pipe(
-      map(user => (isNil(user) ? user : user.name)),
+      map(user => (isNil(user) ? undefined : user.name)),
       publishReplay(1),
       refCount(),
     );
-  }
-
-  /**
-   * Get user
-   */
-  public getUser(token: string = this.getTokenFromStorage()): Promise<UserAuthModel | Observable<never>> {
-    const url = `${this.usersUrl}/${AuthApi.user}`;
-    const body: string = JSON.stringify({
-      token,
-    });
-    const options = {
-      headers: new HttpHeaders({ "Content-Type": "application/json" }),
-    };
-
-    return this.http
-      .post<UserAuthModel>(url, body, options)
-      .toPromise()
-      .then(user => {
-        this.userSubj.next(user);
-        return user;
-      })
-      .catch(err => {
-        this.handleError(err);
-        return throwError("Please try again later");
-      });
+    this.spinner = spinner;
   }
 
   /**
    * Authenticate user
    */
   public authenticate(login: string, password: string): Promise<UserAuthModel | Observable<never>> {
+    this.spinner.show();
```

Comment:
Also, it worth interceptor creation to show spinner on each api call and hide on each response. So, you will need to add spinnerService in one place.